daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/ElementKindParsers.scala

```diff
@@ -119,18 +119,30 @@ class DynamicEscapeSchemeParser(escapeScheme: EscapeSchemeParseEv,
   }
 }
 
-class SequenceCombinatorParser(rd: TermRuntimeData, bodyParser: Parser)
+class SequenceCombinatorParser(rd: TermRuntimeData, childParsers: Array[Parser])
   extends CombinatorParser(rd) {
   override def nom = "Sequence"
 
   override lazy val runtimeDependencies = Nil
 
-  override lazy val childProcessors = Seq(bodyParser)
+  override lazy val childProcessors = childParsers.toSeq
+
+  val numChildParsers = childParsers.size
 
   def parse(start: PState): Unit = {
+    var i: Int = 0
+
     start.mpstate.groupIndexStack.push(1L) // one-based indexing
 
-    bodyParser.parse1(start)
+    while (i < numChildParsers) {
+      val parser = childParsers(i)
+      parser.parse1(start)
+      if (start.processorStatus ne Success) {
+        start.mpstate.groupIndexStack.pop()
+        return
```

Comment:
return's should be avoided. Maybe remove this ``if`` block and instead change the loop to ``while (i > numChildren && start.processorStatus eq Success)``? This has the benefit that it will always execute ``groupIndexStack.pop()`` without duplicating code. Though, I'm not sure if still calling ``moveOverOneGroupIndexOnly`` would break anything, but that could be wrapped in a ``processorStatus eq Success`` check if it matters.